cmd/monitorpartner.go

```diff
@@ -0,0 +1,171 @@
+}
+
+func (d *monitorPartnerDownloader) Do(file interface{}, ds *utils.DoerState) string {
+	var hash string
+	if f, isObject := file.(*vt.Object); isObject {
+		hash = f.ID()
+	} else {
+		hash = file.(string)
+	}
+
+	ds.Progress = fmt.Sprintf("%s %4.1f%%", hash, 0.0)
+
+	// Get download URL
+	var downloadURL string
+	_, err := d.client.GetData(vt.URL("monitor_partner/files/%s/download_url", hash), &downloadURL)
+
+	if err == nil {
+		dstPath := path.Join(viper.GetString("output"), hash)
+		err = d.DownloadFile(downloadURL, dstPath, func(resp *grab.Response) {
+			progress := 100 * resp.Progress()
+			if progress < 100 {
+				ds.Progress = fmt.Sprintf("%s %4.1f%% %6.1f KBi/s",
+					hash, progress, resp.BytesPerSecond()/1024)
+			}
+		})
+	}
+
+	msg := color.GreenString("ok")
+	if err != nil {
+		if apiErr, ok := err.(vt.Error); ok && apiErr.Code == "NotFoundError" {
+			msg = color.RedString("not found")
+		} else {
+			msg = color.RedString(err.Error())
+		}
+	}
+
+	return fmt.Sprintf("%s [%s]", hash, msg)
+}
+
+var monitorPartnerHashDownloadCmdHelp = `Download files from your partner account.
+
+This command download files in your monitor partner account using their sha256.`
+
+var monitorPartnerHashDownloadCmdExample = `  vt monitorpartner download <sha256-1> <sha256-2> ...
+  cat list_of_monitor_ids | vt monitorpartner download -`
+
+// NewMonitorPartnerHashDownloadCmd returns a command for downloading files from your
+// monitor account.
+func NewMonitorPartnerHashDownloadCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "download [sha256]...",
+		Short:   "Download files from your monitor partner account",
+		Long:    monitorPartnerHashDownloadCmdHelp,
+		Example: monitorPartnerHashDownloadCmdExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var argReader utils.StringReader
+			if len(args) == 0 {
+				return errors.New("No hash provided")
+			} else if len(args) == 1 && args[0] == "-" {
+				argReader = utils.NewStringIOReader(os.Stdin)
+			} else {
+				argReader = utils.NewStringArrayReader(args)
+			}
+			client, err := NewAPIClient()
+			if err != nil {
+				return err
+			}
+			re, _ := regexp.Compile("[[:xdigit:]]{64}")
+			monitorHashes := utils.NewFilteredStringReader(argReader, re)
+
+			c := utils.NewCoordinator(viper.GetInt("threads"))
+			c.DoWithStringsFromReader(
+				&monitorPartnerDownloader{fileDownloader: newFileDownloader(client)},
+				monitorHashes)
+			return err
+		},
+	}
+
+	addThreadsFlag(cmd.Flags())
+	addOutputFlag(cmd.Flags())
+	return cmd
+}
+
+var monitorPartnerCmdHelp = `Manage your VirusTotal Monitor Partner account.
+
+This command allows you to list and retrieve files detected by your engine.
+
+Reference:
+  https://developers.virustotal.com/v3.0/reference#monitor-partner`
+
+// NewMonitorPartnerCmd returns a new instance of the 'monitor_hash' command.
+func NewMonitorPartnerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "monitorpartner [sha256]...",
```

Comment:
I think the use string is not correct, `monitorparner` only has subcommands, it doesn't do anything by itself, so why the sha256 argument?

Reply:
right, corrected